syntax = "proto3";

import "map_renderer.proto";
import "transport_router.proto";

package proto_tc;

message Coordinates {
    double lat = 1;
    double lng = 2;
}

message Stop {
    string name = 1;
    Coordinates coordinates = 2;
}

message Bus {
    string name = 1;
    bool is_circular = 2;
    repeated uint32 stops_on_route = 3;
}

message StopsDistance {
    uint32 stop_from = 1;
    uint32 stop_to = 2;
    uint64 distance = 3;
}

message TransportCatalogueData {
    repeated Stop stops = 1;
    repeated Bus buses = 2;
    repeated StopsDistance distances = 3;
}

message TransportCatalogue {
    TransportCatalogueData data = 1;
    RenderSettings render_settings = 2;
    TransportRouter router = 3;
}